### PR TITLE
Feat: add support for absolute paths

### PR DESCRIPTION
**Pull request by rodrigo4244**

### CODE REVIEW

**Comment by rodrigo4244** [comment]
I think that could have a way to DRY a little bit but I don't know how. Another thing is the name of the variables. Would appreciate some feedback!

File: index.js [hunk 1]
```diff
@@ -19,9 +20,13 @@ const assertValidPath = dirPath => {
   }
 }
 
-const listPegjsFiles = rulesDir =>
-  glob.sync('*.pegjs', { cwd: rulesDir, nodir: true })
+const listPegjsFiles = rulesDir => {
+  const pegjsFiles = glob.sync('*.pegjs', { cwd: rulesDir, nodir: true })
     .map(f => path.join(rulesDir, f))
+  if (isEmpty(pegjsFiles)) throw Error('there isnt any .pegjs files in the given path')
```
Comment (gunar) [inline index.js]: Typo: There *aren't*...

Also, please start with a capital letter and end with a full stop.

File: README.md [hunk 2]
```diff
@@ -17,15 +17,15 @@ or
 // src/parsers/index.js
 const mergeParsers = require('@invisible/merge-parsers')
 
-const parsers = mergeParsers({ path: './rules' }) // path contains foo.pegjs and bar.pegjs (src/parsers/rules)
+const parsers = mergeParsers({ path: './rules' }) // path contains foo.pegjs and bar.pegjs (absolute -> src/parsers/rules)
```
Comment (gunar) [inline README.md]: `src/parsers/rules` is not an absolute path.

Absolute paths start with `/`, e.g. `/home/gunar/invisible`.

**Comment by gunar** [comment]
We're adding support for absolute paths, not app-module-path.

**Comment by rodrigo4244** [comment]
I had to rebase because other PR got merged. You can start your review after the commit `docs: fix absolute path`

File: index.js [hunk 3]
```diff
@@ -41,24 +37,20 @@ const makeGraceful = parser => {
 }
 
 const assertDirAndReturnFiles = dirPath => {
-  assertValidPath(dirPath)
-  return listPegjsFiles(dirPath)
+  const valid = assertValidPath(dirPath)
+  return (! valid) ? undefined : listPegjsFiles(dirPath)
```
Comment (gunar) [inline index.js]: Now that the behavior of this function changed I think it is due to receive a new name.

File: index.js [hunk 4]
```diff
@@ -41,24 +37,20 @@ const makeGraceful = parser => {
...
-    absolutePath = buildAbsolutePath(dirPath)
-    filesArray = assertDirAndReturnFiles(absolutePath)
-  } catch (err) {
-    // If the absolute path isnt valid or does not contain any .pegjs files, try a relative path.
+  let relativePath, filesArray
+  filesArray = assertDirAndReturnFiles(dirPath)
+  if (isEmpty(filesArray)) {
+    // If the path isnt valid or does not contain any .pegjs files, try a relative path.
     relativePath = buildRelativePath(dirPath)
     filesArray = assertDirAndReturnFiles(relativePath)
```
Comment (gunar) [inline index.js]: return here to get rid of `let` up there

File: test/index.spec.js [hunk 5]
```diff
@@ -33,6 +33,18 @@ describe('parsers', () => {
...
+    const testPath = __dirname
+
+    assert.throws(() => parsers({ path: testPath }))
+  })
+
+  it('should throw when the relative path given does not contain any .pegjs files', () => {
+    const testPath = './'
+
+    assert.throws(() => parsers({ path: testPath }))
+  })
```
Comment (gunar) [inline test/index.spec.js]: I disagree. It think if the path exists but has no pegjs files in it I would expect this package to simply return an empty object.

File: README.md [hunk 6]
```diff
@@ -17,15 +17,15 @@ or
 // src/parsers/index.js
 const mergeParsers = require('@invisible/merge-parsers')
 
-const parsers = mergeParsers({ path: './rules' }) // path contains foo.pegjs and bar.pegjs (src/parsers/rules)
+const parsers = mergeParsers({ path: './rules' }) // path contains foo.pegjs and bar.pegjs (absolute -> /home/me/myProject/src/parsers/rules)
```
Comment (gunar) [inline README.md]: Remove everything inside the parentheses. Unnecessary. Paths are expected to be either relative or absolute.

File: index.js [hunk 7]
```diff
@@ -6,12 +6,16 @@ const glob = require('glob')
 const path = require('path')
 const peg = require('@invisible/pegjs-import')
 
+const {
+  isEmpty,
+} = require('lodash/fp')
+
 const assertValidPath = dirPath => {
```
Comment (gunar) [inline index.js]: Guess this should be called `isValidPath` now, or smth.

File: package.json [hunk 8]
```diff
@@ -36,6 +35,7 @@
     "@invisible/eslint-config": "^1.1.6",
     "@invisible/publish": "^1.1.8",
     "eslint": "^4.8.0",
+    "lodash": "^4.17.4",
```
Comment (gunar) [inline package.json]: I would have had two separate commits, but ok.

**Pull request merged**